api/src/main/java/org/openmrs/module/appointmentscheduling/api/impl/AppointmentServiceImpl.java

```diff
@@ -1155,6 +1167,117 @@ public Appointment bookAppointment(Appointment appointment,
 				appointment);
 	}
 
+
+	//appointment resource
+
+	@Override
+	public List<AppointmentResource> getAllAppointmentResources() {
+		return getAppointmentResourceDAO().getAll();
+	}
+
+	@Override
+	public List<AppointmentResource> getAllAppointmentResources(boolean includeVoided) {
+		return getAppointmentResourceDAO().getAllData(includeVoided);
+	}
+
+	@Override
+	public AppointmentResource getAppointmentResource(Integer appointmentResourceId) {
+		return (AppointmentResource) getAppointmentResourceDAO().getById(appointmentResourceId);
+	}
+
+	@Override
+	public AppointmentResource getAppointmentResourceByUuid(String uuid) {
+		return (AppointmentResource) getAppointmentResourceDAO().getByUuid(uuid);
+	}
+
+	@Override
+	public AppointmentResource saveAppointmentResource(AppointmentResource appointmentResource) throws APIException {
+		ValidateUtil.validate(appointmentResource);
+		return (AppointmentResource) getAppointmentResourceDAO().saveOrUpdate(appointmentResource);
+	}
+
+	@Override
+	public AppointmentResource voidAppointmentResource(AppointmentResource appointmentResource, String reason) {
+		appointmentResource.setVoided(true);
+		appointmentResource.setVoidReason(reason);
+		return (AppointmentResource) getAppointmentResourceDAO().saveOrUpdate(appointmentResource);
+	}
+
+	@Override
+	public void purgeAppointmentResource(AppointmentResource appointmentResource) {
+		getAppointmentResourceDAO().delete(appointmentResource);
+	}
+
+	@Override
+	public List<AppointmentResource> getAppointmentResourcesByConstraints(Location location, Provider provider, List<AppointmentType> appointmentTypes) {
+		return getAppointmentResourceDAO().getResourceByConstraints(location, provider,null);
+	}
+
+	@Override
+	public TimeSlot getTimeslotForAppointment(Location location, Provider provider, AppointmentType type, Date appointmentDate){
+		TimeSlot requiredTimeslot = null;
+
+		Date dayBeforeAppointmentDate = getDate(appointmentDate, -1);
+		Date dayAfterAppointmentDate = getDate(appointmentDate, 1);
```

Comment:
What is the intent here?  You just want time slots that are available on the current date, correct?

I would think you wouldn't want to get the day before and the day after, but rather the start of the current day and the end of the current date.

That is, if, say the date you pass in is 2018-09-10 then you'd want to call getTimeSlotByConstraints with fromDate="2018-09-10 00:00:00" and toDate=2018-09-10 23:59:59".  Is this not what you are looking for? Or does getTimeSlotsByConstraints not doing what you want?

Reply:
Thanks mark, actually that should work instead of passing a day before and a day after, I could just past fromDate="appointmentDate 00:00:00" and toDate=appointmentDate 23:59:59" to this getTimeSlotsByConstraintsIncludingFull()